conbench/tests/api/test_compare.py

```diff
@@ -336,7 +345,10 @@ def test_compare(self, client):
             batch_id=batch_id,
         )
         new_ids = [e.id for e in new_entities]
-        response = client.get(f"/api/compare/batches/{compare.id}/")
+        query_string = {"threshold_z": threshold_z} if threshold_z else None
+        response = client.get(
+            f"/api/compare/batches/{compare.id}/", query_string=query_string
+        )
```

Comment:
Just a quick comment on HTTP testing style: When I read test code that involves making HTTP requests then right after getting the response I like having something in the lines of:
`assert response.status_code == expected_code, response.text`

This would dump the response contents for the unexpected status code.

The logic that's commented on with ` # cheating by comparing batch to same batch...` only matters when the status code is as expected.

The late ` self.assert_200_ok(response, None, contains=expected[0])` is too indirect/complex for my taste :) I have been trying to do baby steps into 'my preferred style' when I was touching test code here and there. This is no call to action now, just sharing what I do.